### PR TITLE
Added extra logging to workers to see why the won't die

### PR DESCRIPTION
**Pull request by urtubia**
Adding extra logging to see why old workers aren't dying

### CODE REVIEW

File: lib/worker.js [hunk 1]
```diff
@@ -61,9 +61,13 @@ server.listen(function () {
 process.on('message', function (msg) {
   switch (msg.type) {
     case 'die':
+      console.error('worker with pid ' + process.pid + ' received die message');
       setTimeout(function () {
+        console.error('worker with pid ' + process.pid + ' about to exit');
         process.exit(0);
+        console.error('worker with pid ' + process.pid + ' went right by the exit');
```
Comment (bdimcheff) [inline lib/worker.js]: lol

**Comment by urtubia** [comment]
This is what the logs look on console.dev 
https://gist.github.com/urtubia/d2ee64739fec72d8bb54

**Comment by ncrohn** [comment]
So if I am reading the exit method correctly in the docs. Your line after the exit will still run as it will finish the tick that it's processing. I would add that log error in something like below.

```
process.nextTick(function() {
    console.error(Date().toString() + ' -- worker with pid ' + process.pid + ' went right by the exit');
});
```

Another thing you could try is adding an exit listener and logging that it is exiting.
http://nodejs.org/docs/v0.8.0/api/process.html#process_event_exit

**Comment by urtubia** [comment]
@ncrohn great observations, I'll add those.

**Comment by bdimcheff** [comment]
:shipit:

**Pull request merged**